pkg/master/master_test.go

```diff
@@ -531,54 +527,62 @@ func testInstallThirdPartyAPIListVersion(t *testing.T, version string) {
 		},
 	}
 	for _, test := range tests {
-		_, fakeClient, server, assert := initThirdParty(t, version)
-		defer server.Close()
-
-		if test.items == nil {
-			fakeClient.ExpectNotFoundGet(etcdtest.PathPrefix() + "/ThirdPartyResourceData/company.com/foos/default")
-		} else {
-			setupEtcdList(fakeClient, "/ThirdPartyResourceData/company.com/foos/default", test.items)
-		}
+		func() {
+			_, etcdserver, server, assert := initThirdParty(t, version)
+			defer server.Close()
+			defer etcdserver.Terminate(t)
+
+			client := etcdserver.Client
+			if test.items != nil {
+				setupEtcdList(client, "/ThirdPartyResourceData/company.com/foos/default", test.items)
+			}
 
-		resp, err := http.Get(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
-		if !assert.NoError(err) {
-			return
-		}
-		defer resp.Body.Close()
+			resp, err := http.Get(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
+			if !assert.NoError(err) {
+				return
+			}
+			defer resp.Body.Close()
 
-		assert.Equal(http.StatusOK, resp.StatusCode)
+			assert.Equal(http.StatusOK, resp.StatusCode)
 
-		data, err := ioutil.ReadAll(resp.Body)
-		assert.NoError(err)
+			data, err := ioutil.ReadAll(resp.Body)
+			assert.NoError(err)
 
-		list := FooList{}
-		if err = json.Unmarshal(data, &list); err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
+			list := FooList{}
+			if err = json.Unmarshal(data, &list); err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
 
-		if test.items == nil {
-			if len(list.Items) != 0 {
-				t.Errorf("expected no items, saw: %v", list.Items)
+			if test.items == nil {
+				if len(list.Items) != 0 {
+					t.Errorf("expected no items, saw: %v", list.Items)
+				}
+				return
```

Comment:
No - it's fine. It is just exiting the function, which is like "continue in a loop". So that works fine.